### DISCUSSION TITLE
如何在指定field上进行搜索

### DISCUSSION THREAD

**Discussion by secsilm** [milvus-io/milvus#31746] [Q&A and General discussion] [answered]
collection里有两个向量field，我想在搜索的时候在指定field上搜索，比如只在A上进行搜索，目前支持吗？如何做？

**Comment by xiaofan-luan** [comment, marked as answer]
search_param = {
        "data": search_vectors,
        "anns_field": vector_field,
        "param": {"metric_type": _METRIC_TYPE, "params": {"nprobe": _NPROBE}},
        "limit": _TOPK,
        "expr": "id_field >= 0"}
    results = collection.search(**search_param)
    for i, result in enumerate(results):
        print("\nSearch result for {}th vector: ".format(i))
        for j, res in enumerate(result):
            print("Top {}: {}".format(j, res))

**Comment by xiaofan-luan** [comment]
通过指定anns_field，你可以搜索多向量中的某一个field
